src/proxy/outbound.rs

```diff
@@ -104,7 +104,12 @@ impl Outbound {
                             };
                         });
                     }
-                    Err(e) => error!("Failed TCP handshake {}", e),
+                    Err(e) => {
+                        if e.get_ref().unwrap().to_string().contains("shutdown") {
```

Comment:
I think you should compare against the complete error string (i.e. `"A Tokio 1.x context was found, but it is being shutdown."`.

Also, you might want to check if `e.kind() == std::io::ErrorKind::Other` to avoid unnecessary comparisons for other errors.

Reply:
+1. Unfortunate we cannot use `RUNTIME_SHUTTING_DOWN_ERROR` constant directly since it isn't pub

Reply:
I am worried this message could be changed, so I check ket word here

Reply:
LMK if you insist on whole message check

Reply:
or you can add a debug assert that does the slower/less precise check perhaps

Reply:
SG